video/out/cocoa_common.m

```diff
@@ -466,14 +466,21 @@ static CVReturn displayLinkCallback(CVDisplayLinkRef displayLink, const CVTimeSt
     return kCVReturnSuccess;
 }
 
-static void vo_set_level(struct vo *vo, int ontop)
+static void vo_set_level(struct vo *vo, int ontop, int ontop_level)
 {
     struct vo_cocoa_state *s = vo->cocoa;
 
     if (ontop) {
-        // +1 is not enough as that will show the icon layer on top of the
-        // menubar when the application is not frontmost. so use +2
-        s->window_level = NSMainMenuWindowLevel + 2;
+        switch (ontop_level) {
+        case -1:
+            s->window_level = NSFloatingWindowLevel;
+            break;
+        case -2:
+            s->window_level = NSStatusWindowLevel;
+            break;
+        default:
+            s->window_level = ontop_level;
```

Comment:
This is a nice system and easily applicable to other operating systems. I vote for keeping the int-level, because it doesn't hurt anything to allow it.